### ISSUE TITLE
whitelisting of require() external modules does not work when node_modules is under require.root

### ISSUE THREAD

**Issue by jan-osch** [patriksimek/vm2#316]
# Summary
If the `node_modules` folder is under any `require.root` directory, it is possible to load any module located there. 

## Description
My expectation was that the `require.external.modules` mechanism is more strict. If the untrusted code is somehow able to provide a relative or absolute path to a `node_modules` directory, all external modules located there will be exposed. 

When `require.root` is not explicitly provided it allows all paths - so just providing the `require.external.modules` config is not sufficient to block external modules. **I believe that this behavior should be changed or more explicitly stated in the docs**

## Suspected code
I believe that the whitelisting logic should also be applied to the block starting with this line, if a `node_modules` part is detected in the require path: 
https://github.com/patriksimek/vm2/blob/master/lib/sandbox.js#L311

## Example
```
## file structure
vm2-root-external-issue/
    package.json
    index.json
    node_modules/
```

```json
// package.json
{
  "name": "vm2-root-external-issue",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "start": "node index.js"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "fs-extra": "^9.0.1",
    "lodash": "^4.17.19",
    "vm2": "^3.9.2"
  }
}
```

```javascript
// index.js
const { NodeVM } = require('vm2')

const vm = new NodeVM({
  require: {
    external: {
      modules: ['lodash']
    },
    root: [__dirname]
  }
})

vm.run(
  `
      require( './node_modules/fs-extra')
      console.log('external modules whitelisting does not work as expected!')
  `,
  'vm.js'
)
```

### Expected outcome:
The message `external modules whitelisting does not work as expected!` not being printed, a `VMError` thrown.

**Comment by XmiliaH** [comment]
The question is how to handle the module names when an path is given.

**Comment by jan-osch** [comment]
@XmiliaH I can think of two options.

1. In the documentation, we can make it **very clear** that all files under `require.root` directories are accessible by the untrusted code. We could recommend using `require.exernal=false`, a directory only including allowed files as `require.root`, and explicitly pre-loading all external modules to the vm and passing them via `require.mock`. This would be the _recommended way to expose external modules_ in the VM. In that case, we would mark using `require.external.modules` with non-emtpy `require.root` as _advanced usage_ to be used only with extreme care.

2. We can change the `_prepareRequire` implementation and trigger the `require.external.modules` whitelisting logic every time a path with `node_modules` is detected and a non-transitive `require()` is performed. This would mean a breaking change and a version bump.

I would be happy to help with both options. Option 1 is less risky, and probably we should choose that since it is not really a bug, just documentation creating false expectations.

**Comment by XmiliaH** [comment]
Should be fixed in v3.9.7
